src/main/java/frc/robot/util/LogManager.java

```diff
@@ -42,15 +115,21 @@ public static void update() {
    * Records the metadata supplied by gversion (https://github.com/lessthanoptimal/gversion-plugin) in BuildData.java.
    */
   public static void recordMetadata() {
-    new StringLogEntry(DATA_LOG, "BuildData/Maven Group").append(BuildData.MAVEN_GROUP);
-    new StringLogEntry(DATA_LOG, "BuildData/Maven Name").append(BuildData.MAVEN_NAME); // The name of the repository
-    new StringLogEntry(DATA_LOG, "BuildData/Version").append(BuildData.VERSION);
-    new IntegerLogEntry(DATA_LOG, "BuildData/Git Revision").append(BuildData.GIT_REVISION);
-    new StringLogEntry(DATA_LOG, "BuildData/Git SHA").append(BuildData.GIT_SHA); // The SHA code for the latest commit
-    new StringLogEntry(DATA_LOG, "BuildData/Git date").append(BuildData.GIT_DATE);
-    new StringLogEntry(DATA_LOG, "BuildData/Git Branch").append(BuildData.GIT_BRANCH); // The branch name
-    new StringLogEntry(DATA_LOG, "BuildData/Build Date").append(BuildData.BUILD_DATE);
-    new IntegerLogEntry(DATA_LOG, "BuildData/Build Unix Time").append(BuildData.BUILD_UNIX_TIME);
-    new IntegerLogEntry(DATA_LOG, "BuildData/Dirty").append(BuildData.DIRTY);
+    log("BuildData/Maven Group", BuildData.MAVEN_GROUP);
+    log("BuildData/Maven Name", BuildData.MAVEN_NAME); // The name of the repository
+    log("BuildData/Version", BuildData.VERSION);
+    log("BuildData/Git Revision", BuildData.GIT_REVISION);
+    log("BuildData/Git SHA", BuildData.GIT_SHA); // The SHA code for the latest commit
+    log("BuildData/Git date", BuildData.GIT_DATE);
+    log("BuildData/Git Branch", BuildData.GIT_BRANCH); // The branch name
+    log("BuildData/Build Date", BuildData.BUILD_DATE);
+    log("BuildData/Build Unix Time", BuildData.BUILD_UNIX_TIME);
+    log("BuildData/Dirty", BuildData.DIRTY);
+  }
+
+  public enum LogLevel {
+    DEBUG,
+    COMP,
+    NONE
```

Comment:
You have this enum but don't use it. When logging something, you should pass in a log type or use a default log type. Methods in this class should only log it if it is greater than or equal to the log type in Constants.